server/src/main/java/org/opensearch/index/IndexService.java

```diff
@@ -484,11 +491,31 @@ public synchronized IndexShard createShard(
                     warmer.warm(reader, shard, IndexService.this.indexSettings);
                 }
             };
-
+            DiscoveryNode node = targetNode;
             Store remoteStore = null;
-            if (this.indexSettings.isRemoteStoreEnabled()) {
-                Directory remoteDirectory = remoteDirectoryFactory.newDirectory(this.indexSettings, path);
-                remoteStore = new Store(shardId, this.indexSettings, remoteDirectory, lock, Store.OnClose.EMPTY, path);
+            boolean seedRemote = false;
+            if (node.isRemoteStoreNode()) {
+                if (this.indexSettings.isRemoteStoreEnabled()) {
+                    Directory remoteDirectory = remoteDirectoryFactory.newDirectory(this.indexSettings, path);
+                    remoteStore = new Store(shardId, this.indexSettings, remoteDirectory, lock, Store.OnClose.EMPTY, path);
+                } else {
+                    logger.info("Hitting the migration case");
+                    if (sourceNode != null && sourceNode.isRemoteStoreNode() == false) {
+                        assert routing.primary();
+                        logger.info("Hitting the migration case where we need to seed remote store");
+                        seedRemote = true;
+                    }
+                    RemoteSegmentStoreDirectoryFactory directoryFactory = new RemoteSegmentStoreDirectoryFactory(
+                        () -> repositoriesService,
+                        threadPool
+                    );
+                    RemoteSegmentStoreDirectory remoteDirectory = (RemoteSegmentStoreDirectory) directoryFactory.newDirectory(
+                        RemoteStoreNodeAttribute.getRemoteStoreSegmentRepo(this.indexSettings.getNodeSettings()),
+                        this.indexSettings.getUUID(),
+                        shardId
+                    );
+                    remoteStore = new Store(shardId, this.indexSettings, remoteDirectory, lock, Store.OnClose.EMPTY, path);
+                }
```

Comment:
thinking it would be cleaner to use the existing code being used for remote indexes already

Reply:
We can't because it is not remote index already :-)

If we use index settings, it will be used for all other shards as well , which are not migrating , hence we can't do it .

Reply:
Created https://github.com/opensearch-project/OpenSearch/issues/12688